### DISCUSSION TITLE
Default config for projects with AndroidX Compose compiler

### DISCUSSION THREAD

**Discussion by marcin-kozinski** [renovatebot/renovate#20792] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

It's been raised (and answered) a couple times, like https://github.com/renovatebot/renovate/discussions/19082 and https://github.com/renovatebot/renovate/issues/18354, that if your project uses Compose compiler you have to group its updates with Kotlin, otherwise Renovate raises PRs that never work (for Kotlin and for Compose compiler).

It's pretty easy to do it, so it's not a huge blocker (once you figure out this is something you can fix with a config change and either figure it out from documentation or find the buried discussions). But I think it would be a great and easy change to add it to default Renovate config.

Is there any downside or risk in adding this to default config that I'm missing?

I'd be interested in contributing a PR with updates to the default presets.

**Comment by rarkins** [comment]
I think this could be part of `group:recommended`, which is included in `config:base`

**Reply by marcin-kozinski** [reply]
Thanks, makes sense.

Between the two linked discussions, one adds Compose compiler to the "kotlin monorepo" existing group, while the other groups Compose with Kotlin under a new name. Are there any pros and cons to either approach? More specifically for each of the approaches:
* for monorepo: does it make sense to add a related/dependent package to the monorepo group, even though logically it's not part of the monorepo?
* for new group: could it be a problem that Kotlin packages are part of two groups? Could creating a new group impact branch names Renovate creates for Kotlin updates (even in projects without Compose) and is this a problem?

I can try to answer some of these questions on my own (or just choose a solution that works without overthinking), but thought maybe at least some of these questions are going to be easy to answer for maintainers or people more familiar with the project.

**Reply by rarkins** [reply]
It's preferred to avoid adding packages to an existing "monorepo" grouping if it's not actually from that monorepo.

An update will always be part of only one group - last one wins.

**Comment by marcin-kozinski** [comment]
Note, mostly for myself, and for posterity I guess. Not expecting an answer.

I've been testing on a personal project a configuration where I add a package rule to create a new group that groups Kotlin plugin and Compose compiler:
```json
    {
      "matchPackageNames": ["org.jetbrains.kotlin.android", "androidx.compose.compiler:compiler"],
      "groupName": "Kotlin and Compose compiler"
    }
```

This isn't working great. Here are some reasons:
* This changes the branch name for Kotlin only updates from `renovate/kotlin-monorepo` to `renovate/kotlin-and-compose-compiler`. This is okay for my project, where Compose compiler is a dependency. But it doesn't make sense as a default/recommended rule in projects that use Kotlin but not Compose.
* Compose compiler version numbers don't match Kotlin version numbers. So when Renovate finally creates a PR where it updates both it can't extract a version number to put in the PR title, which results in an "immortal" PR. Example: https://github.com/marcin-kozinski/alusia/pull/51

I am going to try some alternative approach. I think I'll start by just disabling Kotlin updates altogether (they can't succeed on their own anyway) and I'll manually update Kotlin on branches Renovate creates for Compose compiler. Ideally I would like Renovate to "know" to update Kotlin to latest any time it updates Compose compiler (but not the other way around), but I don't think this is possible currently.
